packages/vm/lib/buildBlock.ts

```diff
@@ -233,7 +229,13 @@ export class BlockBuilder {
     const blockData = { header: headerData, transactions: this.transactions }
     const block = Block.fromBlockData(blockData, blockOpts)
     await this.vm.blockchain.putBlock(block)
```

Comment:
I generally find it questionable if it is always desired that the generated `Block` is always added to the VM blockchain, I e.g. wonder if this will play well with our current client design (sorry, didn't stumble upon this earlier). Do we might want to at least add a flag `addToBlockchain` or something so that this can be made optional?

Reply:
My reasoning for this is that I wanted the `BlockBuilder` to be ready to build another block as soon as the prior one is finished. I didn't want to the consumer of the API to be responsible for keeping track of what needs to be done in between, preferring for it to be fully self resolving. Also, there are validation checks in `putBlock` that I think are important if they throw then the block wasn't valid from the perspective of the blockchain.

I see what you mean from the perspective of the client though, after having built a block we may not necessarily want to insert it into the blockchain right away?

I know @alcuadrado had to remove his own `putBlock` to accommodate this. @alcuadrado having done that, would you prefer this step remain like this? Or do you think it is a good suggestion from @holgerd77 to add an option for  `addToBlockchain`? Would you prefer it to be default true or false?

Reply:
I found it unexpected, but not critical. The fact that creating a block modifies the state already makes this API generate side effects, so using it to create a block that won't be will always require some special reasoning.

Reply:
Ok, then we can always add such an option later if requested from whatever direction.